src/main/java/uk/gov/hmcts/darts/task/runner/impl/UnstructuredDataExpiryDeletionAutomatedTask.java

```diff
@@ -0,0 +1,167 @@
+import uk.gov.hmcts.darts.log.api.LogApi;
+import uk.gov.hmcts.darts.task.api.AutomatedTaskName;
+import uk.gov.hmcts.darts.task.config.AutomatedTaskConfigurationProperties;
+import uk.gov.hmcts.darts.task.runner.AutoloadingAutomatedTask;
+import uk.gov.hmcts.darts.task.runner.AutoloadingManualTask;
+import uk.gov.hmcts.darts.task.runner.SoftDelete;
+import uk.gov.hmcts.darts.task.service.LockService;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.function.Function;
+
+@Component
+@Slf4j
+public class UnstructuredDataExpiryDeletionAutomatedTask
+    extends AbstractLockableAutomatedTask
+    implements AutoloadingAutomatedTask, AutoloadingManualTask {
+
+    private final UserIdentity userIdentity;
+    private final CurrentTimeHelper currentTimeHelper;
+
+    private final TranscriptionDocumentRepository transcriptionDocumentRepository;
+    private final MediaRepository mediaRepository;
+    private final AnnotationDocumentRepository annotationDocumentRepository;
+    private final CaseDocumentRepository caseDocumentRepository;
+    private final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+
+    private final ExternalInboundDataStoreDeleter inboundDeleter;
+    private final ExternalUnstructuredDataStoreDeleter unstructuredDeleter;
+
+    public UnstructuredDataExpiryDeletionAutomatedTask(
+        AutomatedTaskRepository automatedTaskRepository,
+        AutomatedTaskConfigurationProperties automatedTaskConfigurationProperties,
+        UserIdentity userIdentity,
+        LogApi logApi, LockService lockService,
+        CurrentTimeHelper currentTimeHelper,
+        TranscriptionDocumentRepository transcriptionDocumentRepository, MediaRepository mediaRepository,
+        AnnotationDocumentRepository annotationDocumentRepository,
+        CaseDocumentRepository caseDocumentRepository,
+        ExternalObjectDirectoryRepository externalObjectDirectoryRepository,
+        ExternalInboundDataStoreDeleter inboundDeleter,
+        ExternalUnstructuredDataStoreDeleter unstructuredDeleter) {
+
+        super(automatedTaskRepository, automatedTaskConfigurationProperties, logApi, lockService);
+        this.userIdentity = userIdentity;
+        this.currentTimeHelper = currentTimeHelper;
+        this.transcriptionDocumentRepository = transcriptionDocumentRepository;
+        this.mediaRepository = mediaRepository;
+        this.annotationDocumentRepository = annotationDocumentRepository;
+        this.caseDocumentRepository = caseDocumentRepository;
+        this.externalObjectDirectoryRepository = externalObjectDirectoryRepository;
+        this.inboundDeleter = inboundDeleter;
+        this.unstructuredDeleter = unstructuredDeleter;
+    }
+
+    @Override
+    public AutomatedTaskName getAutomatedTaskName() {
+        return AutomatedTaskName.UNSTRUCTURED_DATA_EXPIRY_DELETION_TASK_NAME;
+    }
+
+    @Override
+    @Transactional
```

Comment:
risky puting this Transactional annotation here, is it needed?
when we hit perf/prod this job could take a while every time its run and lead to table locking and performance issues.

Reply:
Agreed removed :)